sklearn/utils/tests/test_estimator_checks.py

```diff
@@ -404,6 +415,12 @@ def test_check_estimator():
     assert_raises_regex(AssertionError, msg, check_estimator,
                         LargeSparseNotSupportedClassifier)
 
+    # Constant features on bad estimator
+    msg = ("Estimator FailsWithConstantFeatures raised an error due to "
+           "constant features, but does not match expected error message")
+    assert_raises_regex(AssertionError, msg, check_estimator,
```

Comment:
I think we should rather create a new test e.g. `test_check_constant_features()`.

We use `with pytest.raises(Error, match=msg)` now instead of `assert_raises_regex`